### PR TITLE
[Issue #152] feat: Add payments table 

### PR DESCRIPTION
**Pull request by johnkuney**
Adding some UI for the payments page. Made a table component and added transaction data. The new table has better features than the current table on the button pages, so can reuse there in later PR

I just used the dashboard endpoint for the payments page data. Made a few tweaks to get the buttons and transactions in there. Open to other ideas there if thats not ideal

Test plan: go to payments page and test it out 
Note: I added a new package (react-table). The new build seemed to take super long, like 20+ mins...

### CODE REVIEW

File: pages/api/dashboard/index.ts [hunk 1]
```diff
@@ -118,24 +119,29 @@ const getAllMonths = function (transactions: any[]): AllMonths {
 }
 
 const getUserDashboardData = async function (userId: string): Promise<DashboardData> {
-  const buttonsCount = (await paybuttonService.fetchPaybuttonArrayByUserId(userId)).length
+  const buttonsCount = (await paybuttonService.fetchPaybuttonArrayByUserId(userId))
```
Comment (chedieck) [inline pages/api/dashboard/index.ts:122]: Maybe rename it to just `buttons` since is not the count anymore

File: pages/api/dashboard/index.ts [hunk 2]
```diff
@@ -118,24 +119,29 @@ const getAllMonths = function (transactions: any[]): AllMonths {
...
   const addresses = await addressService.fetchAllUserAddresses(userId, true)
   const XECAddresses = addresses.filter((addr) => addr.networkId === XEC_NETWORK_ID)
   const BCHAddresses = addresses.filter((addr) => addr.networkId === BCH_NETWORK_ID)
   const BCHTransactions = Array.prototype.concat.apply([], BCHAddresses.map((addr) => addr.transactions))
   const XECTransactions = Array.prototype.concat.apply([], XECAddresses.map((addr) => addr.transactions))
   const incomingTransactionsInUSD = BCHTransactions.map((t) => {
     t.amount = t.amount.times(DUMMY_BCH_PRICE)
+    t.network = 'BCH'
+    t.button = buttonsCount.find(buttons => buttons.addresses.find(add => add.address.id === t.addressId))
```
Comment (chedieck) [inline pages/api/dashboard/index.ts:131]: This is confusing, `buttons` here refer to a single button

File: pages/api/dashboard/index.ts (hunk 1 above)
Comment (johnkuney) [reply pages/api/dashboard/index.ts]: For sure, added

File: pages/api/dashboard/index.ts (hunk 2 above)
Comment (johnkuney) [reply pages/api/dashboard/index.ts]: For sure, changed to button

**Review by ScottMcDermid — CHANGES_REQUESTED** [review]
bottom-scrollbar

1. Around the < 960px breakpoint, a bottom scrollbar
2. Dark mode's paging buttons' (<< < > >>) have black text on black background.

white-text

3. Light mode's dropdown has light text on light background

**Comment by johnkuney** [comment]
Cool, updated. I was actually seeing the wrong color text on dark mode for the selector lol idk. Was more specific with the css though so hopefully fixed

**Review by ScottMcDermid — APPROVED** [review]
Looks good!

File: pages/api/dashboard/index.ts [hunk 3]
```diff
@@ -118,24 +119,29 @@ const getAllMonths = function (transactions: any[]): AllMonths {
...
+  const buttons = (await paybuttonService.fetchPaybuttonArrayByUserId(userId))
   const addresses = await addressService.fetchAllUserAddresses(userId, true)
   const XECAddresses = addresses.filter((addr) => addr.networkId === XEC_NETWORK_ID)
   const BCHAddresses = addresses.filter((addr) => addr.networkId === BCH_NETWORK_ID)
   const BCHTransactions = Array.prototype.concat.apply([], BCHAddresses.map((addr) => addr.transactions))
   const XECTransactions = Array.prototype.concat.apply([], XECAddresses.map((addr) => addr.transactions))
   const incomingTransactionsInUSD = BCHTransactions.map((t) => {
     t.amount = t.amount.times(DUMMY_BCH_PRICE)
+    t.network = 'BCH'
+    t.button = buttons.find(button => button.addresses.find(add => add.address.id === t.addressId))
```
Comment (chedieck) [inline pages/api/dashboard/index.ts:131]: What if you passed all the buttons? Then in the front end there could be link to all buttons that have the address of the transaction.

You would need only to pass something like
```
t.buttons = buttons.filter(button => button.addresses.some(add => add.address.id === t.addressId)).map((button) => {
  return {
    name: button.name,
    id: button.id
  }
})
```

File: pages/api/dashboard/index.ts [hunk 4]
```diff
@@ -34,6 +34,7 @@ interface DashboardData {
   year: PeriodData
   sevenDays: PeriodData
   all: PeriodData
+  alltransactions: any[]
```
Comment (chedieck) [inline pages/api/dashboard/index.ts:37]: Best to keep it camelCase here: `allTransactions`

File: pages/api/dashboard/index.ts [hunk 5]
```diff
@@ -118,24 +119,29 @@ const getAllMonths = function (transactions: any[]): AllMonths {
 }
 
 const getUserDashboardData = async function (userId: string): Promise<DashboardData> {
-  const buttonsCount = (await paybuttonService.fetchPaybuttonArrayByUserId(userId)).length
+  const buttons = (await paybuttonService.fetchPaybuttonArrayByUserId(userId))
```
Comment (chedieck) [inline pages/api/dashboard/index.ts:122]: The outer parenthesis are now obsolete

File: pages/api/dashboard/index.ts [hunk 6]
```diff
@@ -118,24 +119,29 @@ const getAllMonths = function (transactions: any[]): AllMonths {
...
   const addresses = await addressService.fetchAllUserAddresses(userId, true)
   const XECAddresses = addresses.filter((addr) => addr.networkId === XEC_NETWORK_ID)
   const BCHAddresses = addresses.filter((addr) => addr.networkId === BCH_NETWORK_ID)
   const BCHTransactions = Array.prototype.concat.apply([], BCHAddresses.map((addr) => addr.transactions))
   const XECTransactions = Array.prototype.concat.apply([], XECAddresses.map((addr) => addr.transactions))
   const incomingTransactionsInUSD = BCHTransactions.map((t) => {
     t.amount = t.amount.times(DUMMY_BCH_PRICE)
+    t.network = 'BCH'
```
Comment (chedieck) [inline pages/api/dashboard/index.ts:130]: I think you could not pass this and just use `t.networkId`, comparing it with the constant `XEC_NETWORK_ID` in the file `pages/payments/index.tsx`

**Review by chedieck — COMMENTED** [review]
Looks very nice, left just some minor comments and the idea of showing more buttons if they exist.

**Comment by chedieck** [comment]
Actually just noticed this arrow overlaping with the number: 
image

I'm on firefox, is this something changable?

**Review by Klakurka — APPROVED** [review]
Great v1.

**Pull request merged**